src/main/java/chatbot/cortana/Parser.java

```diff
@@ -0,0 +1,264 @@
+     * @param timeString the date time string to be formatted
+     * @return the formatted date time string
+     */
+    private static String formatDateTimeString(String timeString) {
+        timeString = timeString.trim();
+        timeString = timeString.replace("/", "-");
+        String[] arr = timeString.split(" ");
+        String time = arr[1].trim();
+        String hour = time.substring(0, 2);
+        String minute = time.substring(2, 4);
+        return arr[0] + "T" + hour + ":" + minute + ":" + "00";
+    }
+
+    /**
+     * Parses the date time string to a LocalDateTime object
+     * @param timeString the date time string to be parsed
+     * @return the LocalDateTime object
+     */
+    public static LocalDateTime parseDateTimeString(String timeString) {
+        return LocalDateTime.parse(formatDateTimeString(timeString));
+    }
+
+    /**
+     * Validates the user input
+     * @param command the command to be executed
+     * @param input the user input
+     * @param taskList the list of tasks
+     * @throws InvalidInputException if the input is invalid
+     */
+    public static void validateInput(Command command, String input, TaskList taskList) throws InvalidInputException {
+        if (command == Command.TODO) {
+            if (input.length() <= 4) {
+                throw new InvalidInputException("Please enter a valid todo! "
+                                                + "Tip: todo <description> \nMissing description");
+            }
+        } else if (command == Command.DEADLINE) {
+            if (input.length() > 8) {
+                String[] arr = input.substring(9).split("/by");
+                if (arr.length != 2) {
+                    throw new InvalidInputException("Please enter a valid deadline! "
+                                                    + "Tip: deadline <description> /by <deadline> \nMissing /by");
+                } else {
+                    try {
+                        LocalDateTime.parse(formatDateTimeString(arr[1]));
+                    } catch (Exception e) {
+                        throw new InvalidInputException("Please enter a valid deadline! "
+                                                        + "Tip: deadline <description> /by <deadline>"
+                                                        + "\nInvalid deadline");
+                    }
+                }
+            } else {
+                throw new InvalidInputException("Please enter a valid deadline! "
+                                                + "Tip: deadline <description> /by <deadline> \nMissing description");
+            }
+        } else if (command == Command.EVENT) {
+            if (input.length() > 5) {
+                String[] arr = input.substring(6).split("/from");
+                if (arr.length == 2) {
+                    String[] arr2 = arr[1].split("/to");
+                    if (arr2.length != 2) {
```

Comment:
Consider using more descriptive variable names, and not using numbered variables